testdriver/testplan.py

```diff
@@ -520,18 +520,15 @@ def send_one_line(self, input_line):
                 logging.error(' !!!!!! exec_list = %s\n  input_line = %s' % (self.exec_list, input_line))
                 logging.error(' !!!!!! %s' % self.run_error_message)
 
-                # TODO!!!! Return an error for the offending line instead of failing for the whole batch
-
-                return None
-                input = json.loads(input_line.replace('#EXIT', ''))
-                error_result = {'label': input['label'],
-                                'input_data': input,
+                # Handle problems with decoding errors and other unknowns.
+                error_result = {'label': 'UNKNOWN',
+                                'input_data': input_line,
                                 'error': self.run_error_message
-                }
+                                }
                 return json.dumps(error_result)
         except BaseException as err:
             logging.error('!!! send_one_line fails: input => %s<. Err = %s', input_line, err)
-            input = json.loads(input_line.replace('#EXIT', ''))
+            input = json.loads(input_line.replace('#EXIT', '').strip())
```

Comment:
This seems harmless, but why do we need to replace `#EXIT` from the input string when recording the error? In other words, why aren't we faithfully documenting the input line as it is? I wouldn't expect an `#EXIT` line to be the cause of errors, but even if it is, then our logs should be a faithful source of the evidence.

Reply:
Let's review this later. Tnx for for the LGTM.